charts/mc-router/templates/deployment.yaml

```diff
@@ -0,0 +1,151 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "mc-router.fullname" . }}
+  {{- with .Values.deploymentAnnotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  labels:
+    {{- include "mc-router.labels" . | nindent 4 }}
+  {{- with .Values.deploymentLabels }}
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+spec:
+  {{- if not .Values.autoscaling.enabled }}
+  replicas: {{ .Values.replicaCount }}
+  {{- end }}
+  selector:
+    matchLabels:
+      {{- include "mc-router.selectorLabels" . | nindent 6 }}
+  strategy:
+    {{- .Values.deploymentStrategy | toYaml | nindent 4}}
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "mc-router.labels" . | nindent 8 }}
+        {{- with .Values.podLabels }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "mc-router.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          env:
+            {{- include "mc-router.envBoolMap" (list "IN_KUBE_CLUSTER" "true") | nindent 12 }}
+            {{- include "mc-router.envMap" (list "API_BINDING" (printf ":%d" (.Values.service.apiPort | int))) | nindent 12 }}
+            {{- include "mc-router.envMap" (list "PORT" .Values.service.minecraftPort) | nindent 12 }}
+
+            {{- with .Values.minecraftRouter }}
+              {{- include "mc-router.envBoolMap" (list "AUTO_SCALE_UP" .autoScaleUp.enabled) | nindent 12 }}
+              {{- include "mc-router.envMap" (list "CONNECTION_RATE_LIMIT" .connectionRateLimit) | nindent 12 }}
+              {{- include "mc-router.envMap" (list "CPU_PROFILE" .cpuProfilePath) | nindent 12 }}
+              {{- include "mc-router.envBoolMap" (list "DEBUG" .debug.enabled) | nindent 12 }}
+
+              {{- with .defaultServer }}
+                {{- include "mc-router.envMap" (list "DEFAULT" (printf "%s:%d" .host (.port | int))) | nindent 12 }}
+              {{- end }}
+
+              {{- include "mc-router.envMultilineMap" (list "MAPPING" (include "mc-router.formatMappings" .mappings)) | nindent 12 }}
```

Comment:
Just wanted to confirm that this formats into nothing without any explicit mappings declared?

Reply:
Yes, this is what's generated with the default values:
``` yaml
env:
  - name: IN_KUBE_CLUSTER
    value: "true"
  - name: API_BINDING
    value: ":25580"
  - name: PORT
    value: "25565"
  - name: AUTO_SCALE_UP
    value: "false"
  - name: CONNECTION_RATE_LIMIT
    value: "1"
  - name: DEBUG
    value: "false"
  - name: METRICS_BACKEND
    value: "discard"
  - name: SIMPLIFY_SRV
    value: "false"
  - name: USE_PROXY_PROTOCOL
    value: "false"
  - name: VERSION
    value: "false"
```